schools/api.py

```diff
@@ -217,10 +219,84 @@ class Meta:
                   'archives')
 
 
+class InclusiveFilter(django_filters.Filter):
+    """
+    Filter for including entries where the field is null
+    """
+
+    def filter(self, qs, value):
+        originalqs = super().filter(qs, value)
+        self.lookup_type = 'isnull'
+        nullqs = super().filter(qs, value)
+        return nullqs | originalqs
+
+
+class InclusiveNumberFilter(InclusiveFilter):
+    field_class = forms.DecimalField
+
+
+class NameOrIdFilter(django_filters.Filter):
+    """
+    Filter that switches search target between name and "id", depending on input
+    """
+    table, underscore, column = "", "", ""
+
+    def filter(self, qs, value):
+        if str(value).isdigit():
+            self.field_class = forms.DecimalField
+            if not self.column:
+                # store table and column name
+                self.table, self.underscore, self.column = self.name.rpartition('__')
+            # overwrite column name with column id
+            self.name = self.table + '__id'
+        else:
+            self.field_class = forms.CharField
+            if self.column:
+                # overwrite column id with column name
+                self.name = self.table + '__' + self.column
+        return super().filter(qs, value)
+
+
+class GenderFilter(django_filters.CharFilter):
+    """
+    Filter that maps letters m, f and c to hard-coded genders
+    """
+
+    def filter(self, qs, value):
+        if len(str(value)) == 1:
```

Comment:
Onkohan tässä `str(value)` tarpeellinen? Voisi kuvitella, että sisäänpäin tuleva `value` on aina tyyppiä `str`.

Nyt kun sukupuolelle on oma filtteri, olisi ystävällistä palauttaa virhe, jos devaaja antaa virheellisen arvon:
```python
GENDER_MAP = {
    'm': 'poikakoulu',
    ...
}
val = str(value).lower()
if val not in GENDER_MAP and val not in GENDER_MAP.values():
    raise ParseError("gender must be 'm', 'f' or 'c' (for coed)")
val = GENDER_MAP.get(val, val)
return super().filter(qs, value)
```